dll/win32/imm32/imm.c

```diff
@@ -1795,11 +1795,40 @@ UINT WINAPI ImmGetDescriptionA(
   return len - 1;
 }
 
+#ifdef __REACTOS__ /* FIXME: Don't put on here */
+BOOL WINAPI
+ImmGetImeInfoEx(PIMEINFOEX pImeInfoEx, IMEINFOEXCLASS SearchType, PVOID pvSearchKey);
+#endif
 /***********************************************************************
  *		ImmGetDescriptionW (IMM32.@)
  */
 UINT WINAPI ImmGetDescriptionW(HKL hKL, LPWSTR lpszDescription, UINT uBufLen)
 {
+#ifdef __REACTOS__
+    IMEINFOEX info;
+    WORD w;
+    size_t cchDesc;
+    TRACE("ImmGetDescriptionW(%p, %p, %d)\n", hKL, lpszDescription, uBufLen);
+    if (!ImmGetImeInfoEx(&info, ImeInfoExKeyboardLayout, &hKL))
+        return 0;
+    w = HIWORD(hKL);
+    if ((w & 0xF000) != 0xE000)
+        return 0;
+    cchDesc = wcslen(info.wszImeDescription);
+    if (uBufLen != 0)
+    {
+        if (uBufLen > cchDesc)
+        {
+            wcscpy(lpszDescription, info.wszImeDescription);
+        }
+        else
+        {
+            wcsncpy(lpszDescription, info.wszImeDescription, uBufLen - 1);
+            lpszDescription[uBufLen - 1] = 0;
+        }
+    }
```

Comment:
If you are going to rewrite the whole imm32 ultimately, you could use the strsafe.h functions here.

Reply:
Used `<strsafe.h>`.